src/components/node-list/styles/_filter.scss

```diff
@@ -14,72 +39,89 @@
     fill: none;
     stroke: var(--color-nodelist-filter-indicator-off);
   }
-}
 
-// Fade row text when unchecked
-.pipeline-nodelist__row--unchecked .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.55;
+  .pipeline-nodelist__row:hover &.pipeline-row__toggle-icon--all-unset,
+  [data-whatintent='keyboard'] .pipeline-nodelist__row__checkbox:focus + &,
+  &.pipeline-row__toggle-icon--parent {
+    fill: $color-primary;
+    stroke: $color-primary;
+  }
 }
 
-// Brighter row text when unchecked and hovered
-.pipeline-nodelist__row--unchecked:hover
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.8;
-}
+//--- Toggle icon opacities ---//
 
-// Bright row text when all unset
-.pipeline-nodelist__group--all-unset
-  .pipeline-nodelist__row--unchecked
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 1;
-}
+/*
+ Parent (toggle all tags):
+ | row-hover | icon-hover | checked | opacity | ID |
+ |    🚫     |     🚫      |    🚫   |    0    | 0  |
+ |    ✅     |     🚫      |    🚫   |   0.3   | 1  |
+ |    ✅     |     ✅      |    🚫   |   0.55  | 2  |
+ |    🚫     |     🚫      |    ✅   |   0.55  | 2  |
+ |    ✅     |     ✅      |    ✅   |   0.75  | 3  |
```

Comment:
Love the clarity on this comment

Reply:
It says "Parent (toggle all tags)" as a title and then specifies "opacity" as one of the columns in the comment - what else do you need? I thought it was really obvious.

Reply:
also: ah I understand now! These rules explicitly do not apply to the visibility icons, that’s why they’re on classes for the tag (“filter”) icons only.